### PR TITLE
Run batch processing steps in tasks 

### PR DESCRIPTION
**Pull request by nyalldawson**
Instead of forcing all steps in the batch processing dialog to execute in the main thread, we now run each step as a separate task whenever possible. This keeps the UI nice and responsive, and permits responsive cancelation and progress reporting.

Individual steps are still run sequentially, not in parallel (yet!)

Refs NRCan Contract#3000739399

### CODE REVIEW

**Comment by nyalldawson** [comment]
(This PR also moves most of the guts of the batch execution dialog to c++)

**Review by alexbruy — APPROVED** [review]
Looking good.

File: python/core/auto_generated/processing/qgsprocessingalgrunnertask.sip.in [hunk 1]
```diff
@@ -26,10 +26,13 @@ class QgsProcessingAlgRunnerTask : QgsTask
     QgsProcessingAlgRunnerTask( const QgsProcessingAlgorithm *algorithm,
                                 const QVariantMap &parameters,
                                 QgsProcessingContext &context,
-                                QgsProcessingFeedback *feedback = 0 );
+                                QgsProcessingFeedback *feedback = 0,
+                                bool hidden = false );
```
Comment (3nids) [inline python/core/auto_generated/processing/qgsprocessingalgrunnertask.sip.in:30]: what about using a flag rather than a boolean?

Comment (nyalldawson) [reply python/core/auto_generated/processing/qgsprocessingalgrunnertask.sip.in]: Done

**Comment by agiudiceandrea** [comment]
Please see https://github.com/qgis/QGIS/issues/53806#issuecomment-2764524498.

**Pull request merged**